modules/schema/src/main/java/org/apache/ignite/internal/schema/definition/builder/PrimaryKeyDefinitionBuilderImpl.java

```diff
@@ -98,12 +98,12 @@ public PrimaryKeyDefinition build() {
 
         Set<String> cols = Set.copyOf(columns);
 
-        Set<String> affCols;
+        List<String> affCols;
 
         if (CollectionUtils.nullOrEmpty(affinityColumns)) {
-            affCols = cols;
+            affCols = List.copyOf(columns);
         } else {
-            affCols = Set.copyOf(affinityColumns);
+            affCols = List.copyOf(affinityColumns);
```

Comment:
Earlier we took only unique columns. Should we check that we don't use the same coumns twise?

Reply:
@ygerzhedovich, affinityColumns are checked few lines below.
But affinityColumns are not checked in SchemaConfigurationConverter.